lua/fzfx/cfg/_lsp_locations.lua

```diff
@@ -122,6 +122,7 @@ M._render_lsp_location_line = function(loc)
   if not M._is_lsp_range(range) then
     return nil
   end
+  filename = path.normalize(filename, { double_backslash = true, expand = true })
   if type(filename) ~= "string" or vim.fn.filereadable(filename) <= 0 then
```

Comment:
hi @stong1994 ,

Per your comments:

> Fix some missed references due to unhandled filenames.
For example, let's consider a filename with the value '~/sdk/go1.22.3/src/runtime/runtime-gdb_test.go' at [this location](https://github.com/linrongbin16/fzfx.nvim/blob/c16555686efd6000d14f0f15ad9ca695377b445c/lua/fzfx/cfg/_lsp_locations.lua#L125). When checking its existence using `vim.fn.filereadable(filename)`, the function returns `0`. However, after normalizing the filename, `vim.fn.filereadable(filename)` returns `1`.

The lines are rendered with `~` (or start from `CWD`, current working directory) is because shorter lines look better than full path.

To fix the `vim.fn.filereadable` issue, you can have a try on this API:

```lua
local path = require("fzfx.commons.path")
local normalized_filename = path.normalize(filename, { double_backslash = true, expand = true })
local file_exists = path.isfile(normalized_filename)
```

See document: [path.isfile](https://linrongbin16.github.io/commons.nvim/#/commons_path?id=isfile)

Reply:
Don't worry about the presentation of the filename, as we have formatted it before displaying it @linrongbin16 . And [the code](https://github.com/linrongbin16/fzfx.nvim/blob/c16555686efd6000d14f0f15ad9ca695377b445c/lua/fzfx/cfg/_lsp_locations.lua#L333) is:
```
      local line = string.format(
        "%s:%s:%s:%s",
        providers_helper.LSP_FILENAME_COLOR(vim.fn.fnamemodify(filename, ":~:.")),
        term_color.green(tostring(r.start.line + 1)),
        tostring(r.start.character + 1),
        item_line
      )
```

Reply:
Got it!